packages/material-ui/src/Slider/Slider.test.js

```diff
@@ -0,0 +1,470 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { spy, stub } from 'sinon';
+import { expect } from 'chai';
+import { createMount, getClasses } from '@material-ui/core/test-utils';
+import describeConformance from '@material-ui/core/test-utils/describeConformance';
+import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
+import { cleanup, createClientRender, fireEvent } from 'test/utils/createClientRender';
+import consoleErrorMock from 'test/utils/consoleErrorMock';
+import Slider from './Slider';
+
+function touchList(touchArray) {
+  touchArray.item = idx => touchArray[idx];
+  return touchArray;
+}
+
+function fireBodyMouseEvent(name, properties = {}) {
+  const event = document.createEvent('MouseEvents');
+  event.initEvent(name, true, true);
+  Object.keys(properties).forEach(key => {
+    event[key] = properties[key];
+  });
+  document.body.dispatchEvent(event);
+  return event;
+}
+
+describe('<Slider />', () => {
+  let mount;
+  let render;
+  let classes;
+
+  before(() => {
+    render = createClientRender({ strict: true });
+    classes = getClasses(<Slider value={0} />);
+    mount = createMount({ strict: true });
+  });
+
+  after(() => {
+    cleanup();
+    mount.cleanUp();
+  });
+
+  describeConformance(<Slider value={0} />, () => ({
+    classes,
+    inheritComponent: 'span',
+    mount,
+    refInstanceof: window.HTMLSpanElement,
+    testComponentPropWith: 'span',
+  }));
+
+  it('should call handlers', () => {
+    const handleChange = spy();
+    const handleChangeCommitted = spy();
+
+    const { container, getByRole } = render(
+      <Slider onChange={handleChange} onChangeCommitted={handleChangeCommitted} value={0} />,
+    );
+
+    fireEvent.mouseDown(container.firstChild);
+    document.body.dispatchEvent(new window.MouseEvent('mouseup'));
+
+    expect(handleChange.callCount).to.equal(1);
+    expect(handleChangeCommitted.callCount).to.equal(1);
+
+    fireEvent.keyDown(getByRole('slider'), {
```

Comment:
So I'm not quite sure yet if my approach will be better but the problem with your event is that this will never happen in the browser. Rather the slider will be focused and a keydown event will be fired which prioritizes the active element i.e.

```js
getByRole('slider').focus();
fireEvent.keyDown(document.activeElement)
```
which should increase confidence in the overall behavior of the component because it is closer to how the user would expierence it.